sdk/inc/azure/iot/az_iot_hub_client_properties.h

```diff
@@ -0,0 +1,310 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ *
+ * @brief Definition for the IoT Plug and Play properties writer and parsing routines.
+ */
+
+#ifndef _az_IOT_HUB_CLIENT_PROPERTIES_H
+#define _az_IOT_HUB_CLIENT_PROPERTIES_H
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <azure/core/az_json.h>
+#include <azure/core/az_result.h>
+#include <azure/core/az_span.h>
+
+#include <azure/iot/az_iot_hub_client.h>
+
+#include <azure/core/_az_cfg_prefix.h>
+
+/**
+ * @brief Append the necessary characters to a reported properties JSON payload belonging to a
+ * component.
+ *
+ * The payload will be of the form:
+ *
+ * @code
+ * "reported": {
+ *     "<component_name>": {
+ *         "__t": "c",
+ *         "temperature": 23
+ *     }
+ * }
+ * @endcode
+ *
+ * @note This API only writes the metadata for a component's properties.  The
+ * application itself must specify the payload contents between calls
+ * to this API and az_iot_hub_client_properties_writer_end_component() using
+ * \p ref_json_writer to specify the JSON payload.
+ *
+ * @param[in] client The #az_iot_hub_client to use for this call.
+ * @param[in,out] ref_json_writer The #az_json_writer to append the necessary characters for an IoT
+ * Plug and Play component.
+ * @param[in] component_name The component name associated with the reported properties.
+ *
+ * @pre \p client must not be `NULL`.
+ * @pre \p ref_json_writer must not be `NULL`.
+ * @pre \p component_name must be a valid, non-empty #az_span.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK The JSON payload was prefixed successfully.
+ */
+AZ_NODISCARD az_result az_iot_hub_client_properties_writer_begin_component(
+    az_iot_hub_client const* client,
+    az_json_writer* ref_json_writer,
+    az_span component_name);
+
+/**
+ * @brief Append the necessary characters to end a reported properties JSON payload belonging to a
+ * component.
+ *
+ * @note This API should be used in conjunction with
+ * az_iot_hub_client_properties_writer_begin_component().
+ *
+ * @param[in] client The #az_iot_hub_client to use for this call.
+ * @param[in,out] ref_json_writer The #az_json_writer to append the necessary characters for an IoT
+ * Plug and Play component.
+ *
+ * @pre \p client must not be `NULL`.
+ * @pre \p ref_json_writer must not be `NULL`.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK The JSON payload was suffixed successfully.
+ */
+AZ_NODISCARD az_result az_iot_hub_client_properties_writer_end_component(
+    az_iot_hub_client const* client,
+    az_json_writer* ref_json_writer);
+
+/**
+ * @brief Begin a property response to a writable property request from the service.
+ *
+ * This API should be used in response to an incoming writable properties. More details can be
+ * found here:
```

Comment:
Same here (for a follow-up cleanup PR), does the extra new line after `@brief` let's doxygen know to render it separately? Otherwise, maybe use `@remark` or something else. If it looks fine, then ignore the feedback.

Reply:
Agree.  #1893 assigned to me tracks.